distribution/ddf-common/src/main/resources/bin/start

```diff
@@ -23,17 +23,32 @@
 # ##############################################################################
 
 realpath() {
-  OURPWD=${PWD}
-  cd "$(dirname "${1}")" || exit 2
-  LINK=$(ls -l "$(basename "${1}")" | awk -F"-> " '{print $2}')
-  while [ "${LINK}" ]; do
-	  echo "link: ${LINK}" >&2
-    cd "$(dirname "${LINK}")" || exit 2
+  READLINK_EXISTS=$(command -v readlink &> /dev/null)
```

Comment:
✏️ Another way of formatting this, it doesn't end up shortening the lines at all... but it makes it makes it _easier_ to maintain
```
realpathHelper() {
OURPWD=${PWD}
    cd "$(dirname "${1}")" || exit 2
    LINK=$($2)
    while [ "${LINK}" ]; do
        echo "link: ${LINK}" >&2
        cd "$(dirname "${LINK}")" || exit 2
        LINK=$($2)
    done
    REALPATH="${PWD}/$(basename "${1}")"
    cd "${OURPWD}" || exit 2
    echo "${REALPATH}"
}
realpathHelperReadlinkExist() {
    ls -l "$(basename "${1}")" | awk -F"-> " '{print $2}'
}
realpathHelperReadlinkExtinct() {
    readlink "$(basename "${1}")"
}

realpath() {
  if [ -z $READLINK_EXIST ]; then
     realpathHelper $1 realpathHelperReadlinkExist
  else
     realpathHelper $1 realpathHelperReadlinkExtinct
}
```

Reply:
👍 Got it didn't know that